### PR TITLE
[AD-106] Find solution for Calcite/Janino to resolve root cause of getDefaultCompilerFactory exception.

### PR DESCRIPTION
**Pull request by birschick-bq**
### Summary

[AD-106] Find solution for Calcite/Janino to resolve root cause of getDefaultCompilerFactory exception.

### Description

- Update version of Calcite to 1.28.0 to take advantage of fix for Janino (3.1.6) use.
- Updates to fix warnings for JsonMapper/ObjectMapper.

### Related Issue

https:/bitquill.atlassian.net/browse/AD-106

### Additional Reviewers
@affonsoBQ
@andiem-bq
@alexey-temnikov
@birschick-bq

### CODE REVIEW

File: calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceBasicTest.java [hunk 1]
```diff
@@ -859,14 +859,14 @@ void testComplexQuery() throws SQLException {
                 BsonDocument.parse("{\"$match\": {\"Total\": {\"$gt\": 1}}}"),
                 result.getAggregateOperations().get(5));
         Assertions.assertEquals(
-                BsonDocument.parse(
-                        "{\"$project\": {\"testCollection__id\": \"$testCollection__id\", \"renamed\": \"$field\", \"Total\": \"$Total\", \"_id\": 0}}"),
+                BsonDocument.parse("{\"$sort\": {\"field\": 1}}"),
```
Comment (birschick-bq) [inline calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceBasicTest.java:862]: @andiem-bq Please review the changes to these tests! Do we have an equivalent test against a server for these queries?

File: calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceBasicTest.java [hunk 2]
```diff
@@ -1633,20 +1633,14 @@ void testQueryAndWithTypes() throws SQLException {
...
-                                + "{\"$gt\": [{\"$date\": \"2020-02-01T00:00:00Z\"}, null]}]}, "
-                                + "{\"$lt\": [\"$field\", {\"$date\": \"2020-02-01T00:00:00Z\"}]}, null]}]}]}, true, "
-                                + "{\"$cond\": [{\"$or\": [{\"$eq\": [false, "
-                                + "{\"$cond\": [{\"$and\": [{\"$gt\": [\"$field\", null]}, "
-                                + "{\"$gt\": [{\"$date\": \"2021-01-01T00:00:00Z\"}, null]}]}, "
-                                + "{\"$gt\": [\"$field\", {\"$date\": \"2021-01-01T00:00:00Z\"}]}, null]}]}, "
-                                + "{\"$eq\": [false, {\"$cond\": [{\"$and\": [{\"$gt\": [\"$field\", null]}, "
-                                + "{\"$gt\": [{\"$date\": \"2020-02-01T00:00:00Z\"}, null]}]}, "
-                                + "{\"$lt\": [\"$field\", {\"$date\": \"2020-02-01T00:00:00Z\"}]}, null]}]}]}, false, null]}]}"
+                                + "\"EXPR$0\": {\"$cond\": [{\"$and\": [{\"$eq\": [true, null]}, "
```
Comment (andiem-bq) [inline calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceBasicTest.java:1636]: Concerned about this one and there isn't a server test similar enough to it yet. I don't see the date literals anymore in the new output. testQuerySelectLogicNullAndTypes and testQuerySelectLogicNulls in DocumentDbStatementBasicTest are similar but are not testing the same field against two different literals.

File: calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceBasicTest.java (hunk 1 above)
Comment (andiemontoyeah) [reply calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceBasicTest.java:862]: This particular one looks good to me and is similar to testCompelxQuery in DocumentDbStatementFilterTest.

File: calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceBasicTest.java (hunk 2 above)
Comment (andiemontoyeah) [inline calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceBasicTest.java:1636]: Looks like this got simplified to `EXPR$0=[AND(null, IS NULL(field))` since the condition can never be true. This actually seems correct then but we don't have a test that checks this yet. May need to add a test that checks that the query returns NULL when field is NULL and FALSE when it is not null.

**Review by affonsov — APPROVED** [review]
lgtm

File: calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceBasicTest.java (hunk 2 above)
Comment (birschick-bq) [reply calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceBasicTest.java:1636]: @andiem-bq - Added test in DocumentDbStatementDateTimeTest.java.

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbStatementDateTimeTest.java [hunk 3]
```diff
@@ -1152,6 +1152,41 @@ void testCurrentTimestampMultipleInstances(final DocumentDbTestEnvironment testE
...
+    @DisplayName("Tests select dates with AND condition.")
+    @ParameterizedTest(name = "testSelectDateWithAnd - [{index}] - {arguments}")
+    @MethodSource({"getTestEnvironments"})
+    void testSelectDateWithAnd(final DocumentDbTestEnvironment testEnvironment) throws SQLException {
+        setTestEnvironment(testEnvironment);
+        final String tableName = "testSelectDateWithAnd";
+        final long dateTime = Instant.parse("2020-02-03T04:05:06.00Z").toEpochMilli();
+        final BsonDocument doc1 = BsonDocument.parse("{\"_id\": 101}");
+        doc1.append("field", new BsonDateTime(dateTime));
+        insertBsonDocuments(tableName, new BsonDocument[]{doc1});
```
Comment (andiemontoyeah) [inline src/test/java/software/amazon/documentdb/jdbc/DocumentDbStatementDateTimeTest.java]: Can we add a second document where field does not exist and check for that one the result is NULL instead of FALSE?

Comment (birschick-bq) [reply src/test/java/software/amazon/documentdb/jdbc/DocumentDbStatementDateTimeTest.java]: Added NULL row/result

**Review by andiemontoyeah — APPROVED** [review]
